lib/common/output_messages.c

```diff
@@ -10,36 +10,53 @@
 #include <crm_internal.h>
 #include <crm/common/output_internal.h>
 
-PCMK__OUTPUT_ARGS("attribute", "char *", "char *", "char *")
+PCMK__OUTPUT_ARGS("attribute", "char *", "char *", "char *", "char *")
 static int
 attribute_default(pcmk__output_t *out, va_list args)
 {
     char *scope = va_arg(args, char *);
     char *instance = va_arg(args, char *);
     char *name = va_arg(args, char *);
     char *value = va_arg(args, char *);
+    char *host = va_arg(args, char *);
 
-    if (out->quiet) {
-        pcmk__formatted_printf(out, "%s\n", value);
-    } else {
-        out->info(out, "%s%s %s%s %s%s value=%s",
-                  scope ? "scope=" : "", scope ? scope : "",
-                  instance ? "id=" : "", instance ? instance : "",
-                  name ? "name=" : "", name ? name : "",
-                  value ? value : "(null)");
+    GString *s = g_string_sized_new(50);
+
+    if (scope) {
+        g_string_append_printf(s, "scope=\"%s\" ", scope);
+    }
+
+    if (instance) {
+        g_string_append_printf(s, "id=\"%s\" ", instance);
+    }
+
+    if (name) {
+        g_string_append_printf(s, "name=\"%s\" ", name);
     }
 
+    if (host) {
+        g_string_append_printf(s, "host=\"%s\" ", host);
+    }
+
+    if (value) {
+        g_string_append_printf(s, "value=\"%s\"", value);
+    }
```

Comment:
It should be fine to skip a NULL scope, since crm_attribute likely always passes it and attrd_updater never will, and a NULL instance since crm_attribute already skips that. Empty strings are more questionable but hopefully that's a corner case that most users won't care about.

I would assume name can never be NULL or empty.

Value should be printed as the empty string if NULL or empty, because the value is the point of the query (and there are almost certainly scripts out there relying on that behavior)

Reply:
Updated to fix these.